integration_tests/2pc_test.go

```diff
@@ -1811,12 +1832,26 @@ func (s *testCommitterSuite) TestSetLockedKeyValue() {
 			checkByOpVals(kvrpcpb.Op_Del, []byte{}),
 			checkByOpVals(kvrpcpb.Op_Del, []byte{}),
 		},
+		{
+			"LockAndDeleteYourWrite",
+			[]func(txn transaction.TxnProbe){
+				func(txn transaction.TxnProbe) { txn.SetLockedKeyValue(k1, v1) },
+				func(txn transaction.TxnProbe) { mustLockKey(txn, k1) },
+				func(txn transaction.TxnProbe) {
+					s.Require().NoError(txn.GetMemBuffer().DeleteWithFlags(k1, kv.SetNewlyInserted))
+				},
+			},
+			checkByOpVals(kvrpcpb.Op_Lock, []byte{}),
+			checkByOpVals(kvrpcpb.Op_Lock, []byte{}),
+		},
```

Comment:
We could add cases where `PUT` and `Delete` are executed first then the `SetLockedKeyValue`.